spec/datetimeformat.html

```diff
@@ -282,6 +283,22 @@ <h1>BasicFormatMatcher ( _options_, _formats_ )</h1>
             1. If _format_ has a field [[&lt;_property_&gt;]], let _formatProp_ be _format_.[[&lt;_property_&gt;]]; else let _formatProp_ be *undefined*.
             1. If _optionsProp_ is *undefined* and _formatProp_ is not *undefined*, decrease _score_ by _additionPenalty_.
             1. Else if _optionsProp_ is not *undefined* and _formatProp_ is *undefined*, decrease _score_ by _removalPenalty_.
+            1. Else if _property_ is *"timeZoneName"*, then
+              1. If _optionsProp_ is *"short"* or *"shortGeneric"*, then
+                1. If _formatProp_ is *"shortOffset"*, decrease _score_ by _offsetPenalty_.
+                1. Else if _formatProp_ is *"longOffset"*, decrease _score_ by (_offsetPenalty_ + _shortMorePenalty_).
+                1. Else if _optionsProp_ is *"short"* and _formatProp_ is *"long"*, decrease _score_ by _shortMorePenalty_.
+                1. Else if _optionsProp_ is *"shortGeneric"* and _formatProp_ is *"longGeneric"*, decrease _score_ by _shortMorePenalty_.
+                1. Else if _optionsProp_ ≠ _formatProp_, decrease _score_ by _removalPenalty_.
+              1. Else if _optionsProp_ is *"shortOffset"* and _formatProp_ is *"longOffset"*, decrease _score_ by _shortMorePenalty_.
+              1. Else if _optionsProp_ is *"long"* or *"longGeneric"*, then
+                1. If _formatProp_ is *"longOffset"*, decrease _score_ by _offsetPenalty_.
+                1. Else if _formatProp_ is *"shortOffset"*, decrease _score_ by (_offsetPenalty_ + _longLessPenalty_).
+                1. Else if _optionsProp_ is *"long"* and _formatProp_ is *"short"*, decrease _score_ by _longLessPenalty_.
+                1. Else if _optionsProp_ is *"longGeneric"* and _formatProp_ is *"shortGeneric"*, decrease _score_ by _longLessPenalty_.
+                1. Else if _optionsProp_ ≠ _formatProp_, decrease _score_ by _removalPenalty_.
+              1. Else if _optionsProp_ is *"longOffset"* and _formatProp_ is *"shortOffset"*, decrease _score_ by _longLessPenalty_.
+              1. Else if _optionsProp_ ≠ _formatProp_, decrease _score_ by _removalPenalty_.
```

Comment:
@FrankYFTang this works, but I would love for the adjustments to be expressed more concisely. What would you think of this?
```suggestion
            1. Else if _property_ is *"timeZoneName"*, then
              1. Let _penalty_ be the value of the column with name matching _formatProp_ in the row of <emu-xref href="#table-timezonename-component-non-default-penalties"></emu-xref> for which the value of the &ldquo;Requested Style&rdquo; column is _optionsProp_. If that cell is empty, let _penalty_ be _removalPenalty_.
              1. If _optionsProp_ = _formatProp_, then
                1. Assert: _penalty_ is 0.
              1. Else,
                1. Decrease _score_ by _penalty_.
```
```html
<emu-table id="table-timezonename-component-non-default-penalties">
  <emu-caption>*"timeZoneName"* component non-default penalties</emu-caption>
```

  <table class="real-table">
    <thead>
      <tr>
        <th>Requested Style</th>
        <th>*"short"*</th>
        <th>*"long"*</th>
        <th>*"shortOffset"*</th>
        <th>*"longOffset"*</th>
        <th>*"shortGeneric"*</th>
        <th>*"longGeneric"*</th>
      </tr>
    </thead>
    <tr>
      <td>*"short"*</td>
      <td>0</td>
      <td>_shortMorePenalty_</td>
      <td>_offsetPenalty_</td>
      <td>_offsetPenalty_ + _shortMorePenalty_</td>
      <td></td>
      <td></td>
    </tr>
    <tr>
      <td>*"shortGeneric"*</td>
      <td></td>
      <td></td>
      <td>_offsetPenalty_</td>
      <td>_offsetPenalty_ + _shortMorePenalty_</td>
      <td>0</td>
      <td>_shortMorePenalty_</td>
    </tr>
    <tr>
      <td>*"shortOffset"*</td>
      <td></td>
      <td></td>
      <td>0</td>
      <td>_shortMorePenalty_</td>
      <td></td>
      <td></td>
    </tr>
    <tr>
      <td>*"long"*</td>
      <td>_longLessPenalty_</td>
      <td>0</td>
      <td>_offsetPenalty_ + _longLessPenalty_</td>
      <td>_offsetPenalty_</td>
      <td></td>
      <td></td>
    </tr>
    <tr>
      <td>*"longGeneric"*</td>
      <td></td>
      <td></td>
      <td>_offsetPenalty_ + _longLessPenalty_</td>
      <td>_offsetPenalty_</td>
      <td>_longLessPenalty_</td>
      <td>0</td>
    </tr>
    <tr>
      <td>*"longOffset"*</td>
      <td></td>
      <td></td>
      <td>_longLessPenalty_</td>
      <td>0</td>
      <td></td>
      <td></td>
    </tr>
  </table>
</emu-table>

Reply:
I am nervous about this suggestion. This is a much bigger (pure editorial I assume) changes. How about we merge it w/o such into ECMA402 and then later put that into a different Editorial PR?

Reply:
I don't object either way. But for clarity: would you support such a change, or do you prefer the algorithm steps as they are?

Reply:
Let me put down this way- I am supporting change it to a table form as an editorial change but I think we need more spend more time and review from more people (in particular @anba) for that PR to make sure we didn't change unintentionally. So I prefer not mix that into this Stage 4 PR.